### PR TITLE
No prefetch URLs unless are messages

### PR DESCRIPTION
**Pull request by birkof**

### CODE REVIEW

**Comment by astorije** [comment]
What made you open this?
Did you notice a message type that causes a prefetch? It would be unfortunate that parts/quits trigger a prefetch.

I just tried with a `/me` (`ACTION`) message and it did not prefetch, which is something we should do.

**Comment by birkof** [comment]
I've found that URIs in notice are parsed on server window

**Comment by xPaw** [comment]
@astorije Introduced in 62d4cd8fe824026f37c9564cd6b39f34c78a67de, as it previously only worked in `privmsg`, and it goes via a common function to parse messages, notices, wallops...

**Comment by astorije** [comment]
Gotcha @xPaw. What about allowing `Msg.Type.ACTION` as well?

**Comment by xPaw** [comment]
@astorije Dunno, IRC rfc tends to say that actions should not be acted upon anyhow.

**Comment by astorije** [comment]
Hmm, odd. In terms of UX, as a user, I would expect that "I love that picture: URL" and "/me loves this picture: URL" are being expanded the same in The Lounge.

File: src/plugins/irc-events/message.js [hunk 1]
```diff
@@ -91,6 +91,9 @@ module.exports = function(irc, network) {
 		});
 		chan.pushMessage(client, msg, !self);
 
-		LinkPrefetch(client, chan, msg);
+		// No prefetch URLs unless are simple messages.
+		if (data.type === Msg.Type.MESSAGE) {
```
Comment (astorije) [inline src/plugins/irc-events/message.js]: I would add `Msg.Type.ACTION` here. If strong opinion against this, I'll change my mind, but as a user I'd expect that.

**Comment by astorije** [comment]
@xPaw, @birkof, ping? I really think `Msg.Type.ACTION` should allow for prefetching as well, along with URL parsing, channel parsing, etc.

**Comment by astorije** [comment]
BTW, this is a regression introduced after 2.1.0 so I'm tagging this one as required for 2.2.0 😢

File: src/plugins/irc-events/message.js (hunk 1 above)
Comment (YaManicKill) [inline src/plugins/irc-events/message.js]: Action is "/me", yeah? I agree that prefetch should work with those as well.

**Comment by birkof** [comment]
Hey guys. Sorry for delay. In my opinion, I think it is irrelevant to the user to have notices prefetched on Lobby window (just my 2 cents) that's why i've request this merge

File: src/plugins/irc-events/message.js (hunk 1 above)
Comment (astorije) [inline src/plugins/irc-events/message.js]: Correct.

**Comment by astorije** [comment]
@birkof, I believe we all agree on this :)
`ACTION` represents messages starting with `/me`:

screen shot 2017-01-04 at 21 03 21

Could you add this change here please?
Thanks!

**Comment by astorije** [comment]
@birkof, mind doing this small change so we can put this behind us and get one step closer to the super-near [v2.2.0](https://github.com/thelounge/lounge/milestone/2)? :))

File: src/plugins/irc-events/message.js (hunk 1 above)
Comment (astorije) [inline src/plugins/irc-events/message.js]:
```js
if (data.type === Msg.Type.MESSAGE || data.type === Msg.Type.ACTION) {
```

would probably do the trick.

**Comment by YaManicKill** [comment]
@birkof If you won't have time in the next few weeks, can you let us know, so we can either push it to the next release or update it ourselves. I don't think this is something we want to hold up the release for.

File: src/plugins/irc-events/message.js (hunk 1 above)
Comment (xPaw) [inline src/plugins/irc-events/message.js]: Filter out notices instead?

**Comment by birkof** [comment]
I've also added the ACTION type to prefetching  condition.

**Comment by astorije** [comment]
Woops, looks like you missed your rebase seeing the massive commit list and changes. If you make sure that the "Allow maintainers to make changes" checkbox is checked on the top right section of this page and agree with it, I can fix your branch.

File: src/plugins/irc-events/message.js (hunk 1 above)
Comment (astorije) [inline src/plugins/irc-events/message.js]: @xPaw, I'd rather not. I'd prefer a whitelist than a blacklist, at least at this stage and for v2.2.
That would mean filtering out topics as well probably, parts, quits, aways, etc.

**Comment by xPaw** [comment]
Look at what that file handles, it only affects messages.

**Comment by birkof** [comment]
@xPaw look at message.js file. it handles messages and actions.
image

**Comment by birkof** [comment]
@astorije I allowed edits from maintainers. My mistake was to not make a new branch with this fix, and put it into master.

**Comment by birkof** [comment]
Peeeeerfect! Thanks

**Comment by astorije** [comment]
Hey @birkof, we have sticker packs for our contributors now!
If you're interested, go fill the form at https://goo.gl/forms/f5usqAEp5DWoeXC92 🙂

**Pull request merged**